target/linux/bcm5862x/files/arch/arm/boot/dts/bcm958625-mx64a0.dts

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * Device Tree Bindings for Cisco Meraki MX64(A0).
+ *
+ * Copyright (C) 2021 Matthew Hagan <mnhagan88@gmail.com>
+ */
+
+/dts-v1/;
+
+#include "bcm958625-mx64x.dtsi"
+
+/ {
+	model = "Cisco Meraki MX64(A0)";
+	compatible = "meraki,mx64", "brcm,bcm58625", "brcm,nsp";
+};
+
+&cpu1 {
+	/delete-property/ secondary-boot-reg;
+	secondary-boot-reg = <0xffff042c>;
+};
+
+&L2 {
+	/delete-property/ arm,io-coherent;
+	/delete-property/ prefetch-data;
+	/delete-property/ prefetch-instr;
+};
+
+&amac2 {
+	/delete-property/ dma-coherent;
+};
+
+&ehci0 {
+	/delete-property/ dma-coherent;
+};
+
+&ohci0 {
+	/delete-property/ dma-coherent;
+};
+
+&i2c0 {
+	/delete-property/ dma-coherent;
+};
```

Comment:
Maybe you should not add these if you need to remove them later.

Reply:
Hi @adschm, please refer to "[[PATCH 2/3] ARM: dts: NSP: modify second CPU address](https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg1139904.html)" and [bcm-nsp.dtsi](https://github.com/torvalds/linux/blob/master/arch/arm/boot/dts/bcm-nsp.dtsi) where these properties are defined. The BCM devs have assumed the Ax SoC isn't in use and developed the device tree accordingly, thus these hacks are required to get the board working. I would say the use of /delete-property/ is appropriate here as it highlights this device is a special case. The alternative, perhaps messier, is that we could have a patch remove all instances of dma-coherent and the secondary-boot-reg, then reinsert them in the standard mx64/mx65 dts files along with correct boot-reg in all three.